### PR TITLE
ISPN-8775 Encoder documentation

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.jboss.org/browse/ISPN-8775

### CODE REVIEW

File: core/src/test/java/org/infinispan/dataconversion/GzipEncoder.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+   }
+
+   private byte[] compress(String str) {
+      try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+           GZIPOutputStream gis = new GZIPOutputStream(baos)) {
+         gis.write(str.getBytes("UTF-8"));
+         gis.close();
+         return baos.toByteArray();
+      } catch (IOException e) {
+         throw new RuntimeException("Unable to compress", e);
```
Comment (karesti) [inline core/src/test/java/org/infinispan/dataconversion/GzipEncoder.java]: Won't be better to have a custom runtime exception ?

File: core/src/test/java/org/infinispan/dataconversion/GzipEncoder.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      try (GZIPInputStream gis = new GZIPInputStream(new ByteArrayInputStream(compressed));
+           BufferedReader bf = new BufferedReader(new InputStreamReader(gis, StandardCharsets.UTF_8))) {
+         StringBuilder result = new StringBuilder();
+         String line;
+         while ((line = bf.readLine()) != null) {
+            result.append(line);
+         }
+         return result.toString();
+      } catch (IOException e) {
+         throw new RuntimeException("Unable to decompress", e);
```
Comment (karesti) [inline core/src/test/java/org/infinispan/dataconversion/GzipEncoder.java]: same question as before

File: documentation/src/main/asciidoc/user_guide/encoding.adoc [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+is different.
+
+The data conversions are handled by instances of _org.infinispan.commons.dataconversion.Encoder_ :
+
+[source,java]
+----
+public interface Encoder {
+
+   /**
+    * Convert data in the read/write format to the storage format.
```
Comment (karesti) [inline documentation/src/main/asciidoc/user_guide/encoding.adoc]: is it necessary to include the  javadoc here ?

File: core/src/test/java/org/infinispan/dataconversion/GzipEncoder.java (hunk 1 above)
Comment (karesti) [reply core/src/test/java/org/infinispan/dataconversion/GzipEncoder.java]: I didn't see that this is in test

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**